core/src/main/java/feast/core/service/JobCoordinatorService.java

```diff
@@ -177,16 +158,145 @@ private void updateFeatureSetStatuses(List<JobUpdateTask> jobUpdateTasks) {
     featureSetRepository.flush();
   }
 
+  /**
+   * Makes Job Update Tasks required to reconcile the current ingestion jobs with the given source
+   * to store map. Compares the current ingestion jobs and source to store mapping to determine
+   * which whether jobs have started/stopped/updated in terms of Job Update tasks. Only tries to
+   * stop ingestion jobs its the required ingestions to maintained ingestion jobs are already
+   * RUNNING.
+   *
+   * @param sourceStoreMap defining the source to store pairs where ingestion jobs would have to be
+   *     maintained for ingestion to work correctly.
+   * @return list of job update tasks required to reconcile the current ingestion jobs to the state
+   *     that is defined by sourceStoreMap.
+   */
+  private List<JobUpdateTask> makeJobUpdateTasks(Map<Source, Store> sourceStoreMap) {
+    List<JobUpdateTask> jobUpdateTasks = new ArrayList<>();
+    // Ensure a running job for each source to store mapping
+    long updateTimeout = jobProperties.getJobUpdateTimeoutSeconds();
+    List<String> requiredJobsIds = new ArrayList<>();
+    boolean isSafeToStopJobs = true;
+    for (Map.Entry<Source, Store> entry : sourceStoreMap.entrySet()) {
+      Source source = entry.getKey();
+      Store store = entry.getValue();
+
+      Optional<Job> sourceToStoreJob = getJob(source, store);
+      if (sourceToStoreJob.isPresent()) {
+        Job job = sourceToStoreJob.get();
+        // Record the job id as required to safeguard it from getting stopped.
+        requiredJobsIds.add(sourceToStoreJob.get().getId());
+        if (!job.isRunning()) {
+          // Mark that it is not safe to stop jobs without disrupting ingestion
+          isSafeToStopJobs = false;
+        }
+      }
+
+      jobUpdateTasks.add(
+          new JobUpdateTask(
+              getFeatureSetsForStore(store),
+              source,
+              store,
+              sourceToStoreJob,
+              jobManager,
+              updateTimeout,
+              JobStatus.RUNNING));
+    }
+    // Stop extra jobs that are not required to mantain ingestion when safe
+    if (isSafeToStopJobs) {
+      getExtraJobs(requiredJobsIds)
```

Comment:
This can generate pretty big SQL query by putting long list of Ids. I'm not sure what's the length limit as, but definitely not the best practice.
I suggest to have 
```
Set<Job> allRunningJobs = getAllRunningJobs();
Set<Job> checkedAsNeeded;
for (pair : sourceToStorePairs) {
   checkedAsNeeded.add(..)
}
Set<Job> toStop = Sets.difference (allRunningJobs, checkedAsNeeded);
```

Reply:
Updated `getExtraJobs()` do the diff in memory instead of as an SQL query